lib/utils/houchdb.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+const PouchW = require('pouchdb-wrapper');
+const _ = require('lodash');
+const defaultOptions = {
+    namespace: 'default',
+    port: 5984,
+    hostname: 'localhost',
+    protocol: 'http',
+    db: 'my-db',
+    plugins: []
+};
+
+const register = function(server, options, next) {
+    options = _.merge(defaultOptions, options);
+
+    // register PouchDB plugins
+    // this appears safe to do repeatedly, every time we call this function
+    _.forEach(options.plugins, (plugin) => {
+        PouchW.PouchDB.plugin(plugin);
+    });
+
+    let config;
+    if (options.path) {
+        config = {
+            path: options.path,
+            name: options.name
+        };
+    } else if (options.conn) {
+        config = {
+            name: options.name,
+            conn: options.conn
+        };
+    }
+
+    const db = new PouchW(config);
+    const pluginName = module.exports.attributes.name;
+    const exposableObj = {};
+    exposableObj[options.name] = db;
+    exposableObj.constructor = PouchW;
+
+    // throw error if namespace already exists
+    if (
+        server.plugins[pluginName] &&
+        server.plugins[pluginName][options.name]
+    ) {
+        const msg = 'PouchDB namespace `' + options.prefix + '` already exists';
+        next(new Error(msg));
+    }
+
+    // attempt to get info from the db
+    return db.info()
```

Comment:
I C. a `.save({_id: '_upsertdb'}).then(doc => db.delete(doc))` is a pronto hack.  otherwise we'd just need to test once against `GET  remoteDbUrlRoot + '/dbname'` for a  `404` then follow up with a `POST remoteDbUrlRoot + '/dbname'` if so